lib/common.sh

```diff
@@ -268,132 +326,144 @@ is_private_ip()
 }
 
 #
-# Mount helper must call this function to grab a timed lease on all MOUNTMAP
+# Mount helper must call this function to grab a timed lease on all MOUNTMAPv3
 # entries. It should do this if it decides to use any of the entries. Once
-# this is called aznfswatchdog is guaranteed to not delete any MOUNTMAP till
+# this is called aznfswatchdog is guaranteed to not delete any MOUNTMAPv3 till
 # the next 5 minutes.
 #
-# Must be called with MOUNTMAP lock held.
+# Must be called with MOUNTMAPv3 lock held.
 #
-touch_mountmap()
+touch_mountmapv3()
 {
-    chattr -f -i $MOUNTMAP
-    touch $MOUNTMAP
+    chattr -f -i $MOUNTMAPv3
+    touch $MOUNTMAPv3
     if [ $? -ne 0 ]; then
-        chattr -f +i $MOUNTMAP
-        eecho "Failed to touch ${MOUNTMAP}!"
+        chattr -f +i $MOUNTMAPv3
+        eecho $LOG_V3 "Failed to touch ${MOUNTMAPv3}!"
         return 1
     fi
-    chattr -f +i $MOUNTMAP
+    chattr -f +i $MOUNTMAPv3
+}
+
+# Create NFS V4 mount map file
+touch_mountmapv4()
+{
+    if [ ! -f $MOUNTMAPv4 ]; then
+        touch $MOUNTMAPv4
+        if [ $? -ne 0 ]; then
+            eecho $LOG_V4 "[FATAL] Not able to create '${MOUNTMAPv4}'!"
+            exit 1
+        fi
+    fi
 }
```

Comment:
Is this intentional that mountmapv4 is not immutable? If not, I would suggest to make it immutable to avoid it being modified by any user unintentionally.

Reply:
Fixed.